cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AdapterHttpHandler.java

```diff
@@ -276,7 +276,7 @@ public void deleteAdapter(HttpRequest request, HttpResponder responder,
       adapterService.removeAdapter(Id.Namespace.from(namespaceId), adapterName);
       responder.sendStatus(HttpResponseStatus.OK);
     } catch (CannotBeDeletedException e) {
-      responder.sendString(HttpResponseStatus.FORBIDDEN, e.getMessage());
+      responder.sendString(HttpResponseStatus.CONFLICT, e.getMessage());
     } catch (NotFoundException e) {
```

Comment:
We should probably do a walk through all the handlers and remove unnecessary catch clauses. I created https://issues.cask.co/browse/CDAP-2422. @jwang47 can you add some more details to that, please?